### ISSUE TITLE
Import bug in Jest - only in tests

### ISSUE THREAD

**Issue by pstachula-dev** [lucide-icons/lucide#1525]
## Prerequisites

Version: Release v0.268.0
Running from: Import using webpack
Operating system: Linux
Bits: 64 bits
Next.js v13.4.14

## Step to reproduce

In my case upgrade Next.js doesn't help - https://github.com/lucide-icons/lucide/issues/1482

I have this issue only in Jest when I am running tests:

```
 Cannot find module 'modularize-import-loader?name=X&from=default&as=default&join=../esm/icons/x!lucide-react'
```

My jest config:
```
const customJestConfig = {
  moduleNameMapper: {
    '^@/design-system/(.*)$': '<rootDir>/src/design-system/$1',
    '^@/app/(.*)$': '<rootDir>/src/app/$1',
  },
  setupFilesAfterEnv: ['<rootDir>/jest.setup.js'],
  testEnvironment: 'jest-environment-jsdom',
};
```

**Comment by ericfennis** [comment]
@pstachula-dev Thanks for reporting this.
Can you provide a reproduction repo?

**Comment by donatoaguirre24** [comment]
I'm having the same issue after updating to `"next": "13.4.19"` and `"lucide-react": "0.269.0"`

**Comment by JanTrichter** [comment]
@ericfennis could reproduce the issue [in this repo](https://github.com/JanTrichter/lucide-next-jest-bug):

```bash
git clone https://github.com/JanTrichter/lucide-next-jest-bug.git
```

and then cd into the project and run 

```bash
npm test
```

run all test and then you should see:

```
Cannot find module 'modularize-import-loader?name=CameraIcon&from=default&as=default&join=../esm/icons/camera!lucide-react' from 'pages/index.tsx'
```

Checked another project (`react-feather`) and it seems to be lucide that has an issue here and not some general problem.

**Comment by ericfennis** [comment]
@JanTrichter Thanks for the reproduction repo. I looked at it and I concluded this is a Nextjs problem.
Nextjs had some hardcoded "modularize imports" for `lucide-react` in their repo. This caused some issues before: See #1482

This issue for Jest, looks like they fixed it in the latest canary release.

If you install the canary version in your reproduction repo  `npm install next@canary` and run the tests again, you see that the tests are working as expected.

I close this one as completed.
